docs/ci-cd-environment/docker-authentication.md

```diff
@@ -0,0 +1,157 @@
+```bash
+sem create secret <name-of-your-secret> \
+  -e DOCKER_CREDENTIAL_TYPE=DockerHub \
+  -e DOCKER_USERNAME=<your-dockerhub-username> \
+  -e DOCKER_PASSWORD=<your-dockerhub-password> \
+```
+**Adding a secret to your pipeline YAML**  
+To use your newly created secret in your jobs you have to first attach it.  
+You can attach a secret to individual blocks in your workflow or the whole pipeline.  
+We suggest doing the latter so that it's available to **all jobs** in the workflow.  
+You can achieve this by using `global_job_config` like this:  
+```yaml
+version: v1.0
+name: My project
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu1804
+
+global_job_config:
+  # Connect secret to all jobs in the pipeline
+  secrets:
+    - name: <your-docker-hub-secret>
+
+blocks:
+  ...
+```
+
+### Use secret to authenticate Docker images pulls  
+For your docker pulls to be authenticated you have to log into Docker Hub:  
+```bash
+echo $DOCKER_PASSWORD | docker login --username "$DOCKER_USERNAME" --password-stdin
+```
+You should run this command before pulling any Docker images.  
+
+Same as with secrets, to avoid having to add this to every job or block we suggest that you include it in the `global_job_config` prologue of your pipeline:
+```yaml
+global_job_config:
+  prologue:
+  # Execute at the start of every job in the pipeline
+    commands:
+      - echo $DOCKER_PASSWORD | docker login --username "$DOCKER_USERNAME" --password-stdin
+  ...
+```
+This way the `docker login` command will be run at the start of each job in the pipeline.
+
+If however, you prefer to log in to Docker Hub in individual jobs only, you can do it like in this example:
+```bash
+# .semaphore/semaphore.yml
+version: v1.0
+name: Using a Docker image
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu1804
+
+blocks:
+  - name: Run container from Docker Hub
+    task:
+      jobs:
+      - name: Authenticate docker pull
+        commands:
+          - checkout
+          - echo $DOCKER_PASSWORD | docker login --username "$DOCKER_USERNAME" --password-stdin
+          - docker pull <repository>/<image>
+          - docker images
+          - docker run <repository>/<image>
+      secrets:
+      - name: docker-hub
+```
+
+
+### Running jobs inside a Docker image
+When you're using Docker image as your pipeline CI/CD environment make sure to attach the `docker-hub` secret to your agent's properties to pull the images:
+```bash
+agent:
+  machine:
+    type: e1-standard-2
+
+  containers:
+    - name: main
+      image: <repository>/<image>
+
+  image_pull_secrets:
+    - name: <your-docker-hub-secret>
+```
+
+## How to check if you are logged in
+The `docker login` command will display a **Login Succeeded** message as an output if authentication was successfull. 
+
+Another way to check is to open `~/.docker/config.json` and check the `auths` field. 
+
+If you have been successfully logged in then the `auths` field will be updated accordingly:
```

Comment:
If you have been successfully logged in then the `auths`  => If you have been successfully logged in, then the `auths`